File: src/main/java/io/bloco/faker/components/Company.java
package io.bloco.faker.components;

import java.util.List;

import io.bloco.faker.FakerComponent;
import io.bloco.faker.FakerData;

public class Company extends FakerComponent {

    public Company(FakerData data) {
        super(data);
    }

    public String name() {
        return parse(fetch("company.name"));
    }

    public String suffix() {
        return fetch("company.suffix");
    }

    public String industry() {
        return fetch("company.industry");
    }

    public String catchPhrase() {
        List buzzwordsSections = getList("company", "buzzwords");
        String catchPhrase = "";
        for (Object buzzwordsSection : buzzwordsSections) {
            catchPhrase += sampleFromList((List) buzzwordsSection) + " ";
        }
        return catchPhrase.substring(0, catchPhrase.length() - 1);
    }

    public String buzzwords() {
        return fetch("company.buzzwords");
    }

    public String bs() {
        return fetch("company.bs");
    }

    public String logo() {
        int randomNum = randomHelper.range(1, 12);
        return "https://pigment.github.io/fake-logos/logos/medium/color/" + randomNum + ".png";
    }

    public String profession() {
        return fetch("company.profession");
    }
}

File: src/main/kotlin/io/bloco/faker/components/Company.kt
package io.bloco.faker.components

import io.bloco.faker.FakerComponent
import io.bloco.faker.FakerData
import io.bloco.faker.helpers.RandomHelper

class Company(data: FakerData) : FakerComponent(data) {
    fun name(): String {
        return parse(fetch("company.name"))
    }

    private fun prefix(): String {
        return fetch("company.prefix")
    }

    fun suffix(): String {
        return fetch("company.suffix")
    }

    fun industry(): String {
        return fetch("company.industry")
    }

    fun catchPhrase(): String {
        val buzzwordsSections = getList("company", "buzzwords")
        return buzzwordsSections.joinToString(" ") { buzzwordSection ->
            sampleFromList(buzzwordSection as List<*>)
        }
    }

    fun buzzwords(): String {
        return fetch("company.buzzwords")
    }

    fun bs(): String {
        return fetch("company.bs")
    }

    fun logo(): String {
        val randomNum = RandomHelper.range(1, 12)
        return "https://pigment.github.io/fake-logos/logos/medium/color/$randomNum.png"
    }

    fun profession(): String {
        return fetch("company.profession")
    }
}
